source/elements/oneDAL/source/common_interface/error_handling.rst

```diff
@@ -1,3 +1,79 @@
+Only leaf classes are inherited from the corresponding STL exception classes and could be thrown.
+
+**Inheritance:**
+
+::
+
+    class onedal::exception;
+    class onedal::logic_error : public onedal::exception;
+    class onedal::runtime_error : public onedal::exception;
+
+    class onedal::invalid_argument :  public onedal::logic_error, public std::invalid_argument;
+    class onedal::domain_error :  public onedal::logic_error, public std::domain_error;
+    class onedal::out_of_range :  public onedal::logic_error, public std::out_of_range;
+    class onedal::unimplemented_error :  public onedal::logic_error, public std::logic_error;
+    class onedal::unavailable_error :  public onedal::logic_error, public std::logic_error;
+
+    class onedal::range_error :  public onedal::runtime_error, public std::range_error;
+    class onedal::overflow_error :  public onedal::runtime_error, public std::overflow_error;
+    class onedal::system_error :  public onedal::runtime_error, public std::system_error;
+    class onedal::internal_error :  public onedal::runtime_error, public std::runtime_error;
+
+    class onedal::bad_alloc :  public onedal::exception, public std::bad_alloc;
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Exception
+     - Description
+   * - onedal::exception
+     - It is abstract base class for all throwing exceptions.
+   * - onedal::logic_error
+     - It is abstract class and reports errors that are a consequence of faulty
+       logic within the program such as violating logical preconditions or class
+       invariants and may be preventable.
+   * - onedal::runtime_error
+     - It is abstract class and reports errors that are due to events beyond the scope of the program and can not be easily predicted.
+   * - onedal::invalid_argument
+     - It reports errors that arise because an argument value has not been accepted.
+   * - onedal::domain_error
+     - It reports situations where the inputs are outside of the domain on which an operation is defined.
+   * - onedal::out_of_range
+     - It reports errors that are consequence of attempt to access elements out of defined range.
+   * - onedal::unimplemented_error
+     - It reports errors that arise because an algorithm or method is not implemented.
+   * - onedal::unavailable_error
+     - It reports situations when a device or data is not available.
+   * - onedal::range_error
+     - It reports situations where a result of a computation cannot be represented by the destination type.
+   * - onedal::overflow_error
+     - It reports arithmetic overflow errors (that is, situations where a result of a computation is too large for the destination type)
+   * - onedal::system_error
+     - It reports exceptions that have an associated error_code.
+   * - onedal::internal_error
+     - It reports all runtime errors that couldn't be assigned to other inheritors.
+   * - onedal::bad_alloc
+     - It reports failure to allocate storage.
```

Comment:
for the rest of exceptions, I'd add a new column with its type (logic error, runtime error)
plus it makes sense to sort alphabetically for each type of error